scripts/kernel_installer.py

```diff
@@ -0,0 +1,170 @@
+    shell=True,
+    cwd=tempesta_path
+    )
+output, err = p_kern_hash.communicate()
+current_patch_hash = output.decode("utf-8").strip()[:7]
+current_kernel_hash = os.uname().release.strip().split('-')[-1][:7]
+
+if current_patch_hash == current_kernel_hash:
+    print('Same kernel version installed')
+    sys.exit(0)
+else:
+    print('New kernel version found')
+    # Step 0: Check if kernel already present in repo
+    with subprocess.Popen(["apt-get", "update"],
+                          stdout=subprocess.PIPE) as p:
+        for line in p.stdout:
+            print(line, end='\n') # process line here
+
+    with subprocess.Popen(["apt-cache", "search", f"{current_patch_hash}"],
+                          stdout=subprocess.PIPE) as packages:
+        # If present in repo - install from apt
+        if len(packages.stdout.readlines()) >= 3:
+            print('Kernel found, installing from repo')
+            with subprocess.Popen(["apt", "install", f"*5.10.35.tfw-{current_patch_hash}"],
+                          stdout=subprocess.PIPE) as p:
+                for line in p.stdout:
+                    print(line, end='\n')
+
+        # Else - Run local build
+        else:
+            print('Need to get another version of kernel')
+            print(f'{current_patch_hash} != {current_kernel_hash}')
+            
+            
+
+            # Step 1: Download sources
+            print('Step 1: Downloading 5.10.35 sources...')
+            url = 'https://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git/snapshot/linux-5.10.35.tar.gz'
+            r = requests.get(url, allow_redirects=True)
+            open('download.tar.gz', 'wb').write(r.content)
+            print('Done')
+
+            # Step 2: Untar
+            print('Step 2: Untar sources...')
+            my_tar = tarfile.open('download.tar.gz')
+            my_tar.extractall('.') # specify which folder to extract to
+            my_tar.close()
+            print('Done')
+
+            # Step 3: Apply EXTRAVERSION and patch
+            print('Step 3: Patch...')
+            with open('linux-5.10.35/Makefile') as f:
+                    data = f.read()
+                    data = data.replace('EXTRAVERSION =', f'EXTRAVERSION = .tfw-{current_patch_hash}')
+                    data = data.replace('NAME = Dare mighty things', f'NAME = Some test string')
+            with open(r'linux-5.10.35/Makefile', 'w') as file:
+                file.write(data)
+            shutil.copyfile(f'{tempesta_path}/linux-5.10.35.patch', 'linux-5.10.35/linux-5.10.35.patch')
+            with subprocess.Popen(["patch -p1 < linux-5.10.35.patch"],
+                                  stdout=subprocess.PIPE,
+                                  shell=True,
+                                  cwd=f"linux-5.10.35") as p:
+                output, err = p.communicate()
+                patch_out = output.decode("utf-8").strip()
+
+            # Step 4: config
+            if os.path.exists('.config'):
```

Comment:
When downloading vanilla sources, there won't be a `.config` in an archive. `.config` is basically a result of kernel configuration, e.g. someone called `make menuconfig`. Which `.config` are you referring to here?

Reply:
I add that functional in case somebody can use this script to build from original non-tempesta kernel. And just if .config file is exists in current location - it will be used. 
Just in case